src/index.ts

```diff
@@ -265,12 +265,21 @@ export class Paginator {
     const callback = parsedArguments.callback!;
     if (parsedArguments.autoPaginate) {
       const results = new Array<{}>();
-      paginator.runAsStream_(parsedArguments, originalMethod)
-          .on('error', callback)
-          .on('data', (data: {}) => results.push(data))
-          .on('end', () => callback(null, results));
+      if (callback) {
```

Comment:
Can we just make one call to `runAsStream_` wrapped in a promise and if we detect a callback do something like 

```js
promise.then(results => callback(null, ...results), callback);
```

Reply:
Thanks, changed the code accordingly. PTAL.